Source/JavaScriptCore/API/PASReportCrashPrivate.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef PASReportCrashPrivate_hpp
+#define PASReportCrashPrivate_hpp
+
+#include <stdio.h>
+#import <JavaScriptCore/JavaScriptCore.h>
+#import <JavaScriptCore/JSExportMacros.h>
+#include <bmalloc/pas_report_crash.h>
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if !PLATFORM(WATCHOS)
```

Comment:
This is a different guard than what libpas uses:
```
#if PAS_OS(DARWIN)
```

Is that ok? It seems like non-Darwin ports would compile this function when `pas_report_crash_extract_pgm_failure` is not available.

Reply:
I think we need both.

WATCHOS uses bmalloc